drivers/soundwire/mipi_disco.c

```diff
@@ -168,6 +168,20 @@ static int sdw_slave_read_dp0(struct sdw_slave *slave,
 	dp0->imp_def_interrupts = fwnode_property_read_bool(port,
 				"mipi-sdw-imp-def-dp0-interrupts-supported");
 
+	nval = fwnode_property_count_u32(port, "mipi-sdw-lane-list");
+	if (nval > 0) {
+		dp0->num_lanes = nval;
+		dp0->lane_list = devm_kcalloc(&slave->dev,
+					      dp0->num_lanes, sizeof(*dp0->lane_list),
+					      GFP_KERNEL);
+		if (!dp0->lane_list)
+			return -ENOMEM;
+
+		fwnode_property_read_u32_array(port,
```

Comment:
No error check? What can go wrong?.. :-)

Reply:
well, this is a specific case @andy-shev. We first read the array to figure out its element count, then read a second time to copy the contents in the allocated array. The existing code for SoundWire assumes the second read completes without any errors, and for consistency I used the same programming pattern.

If this is wrong then we need to correct the entire file...

Reply:
This __might__ be wrong. Basically there is no guarantee that the second read will succeed. For example, `asprinf()`-based functions in kernel have a specific warning in case the string (length) had been changed between two calls. It means, there is no guarantee that similar never happen here. TL;DR: I suggest considering such a change.

Reply:
yeah, but here we're reading fixed information set in stone in the DSDT... The DisCo stuff is so horrible no one in their right mind will try to do something dynamic. The DisCo tool generate a file that's not supposed to be modified, ever.

Reply:
I understand that, but what I'm talking about is the OS part (data corruption, other thingy), which DisCo tool and ACPI has no control over. Additionally, but I believe it's pure theoretical, some of that might be instantiated via SSDT overlay, which **is** dynamic ACPI piece of namespace.